bot/cogs/dm_comms.py

```diff
@@ -0,0 +1,59 @@
+import discord
+from discord import DMChannel, Interaction, Message
+from discord.ext import commands
+from discord.ui import Button, button
+from rodhaj import Rodhaj
+
+
+class StubView(discord.ui.View):
+    def __init__(self, base_message: Message):
+        super().__init__(timeout=30)
+        self._base_msg = base_message
+        self._replied = False
+
+    async def on_timeout(self) -> None:
+        if not self._replied:
+            await self._base_msg.reply("You have not responded within 30 seconds")
+
+    @button(label="Create Thread")
+    async def create_thread_btn(self, interaction: Interaction, btn: Button):
+        if not self._replied:
+            self._replied = True
+            btn.disabled = True
+            # thread code creation goes to here
+            await interaction.message.edit(content="Thread created!", view=self)
+            await interaction.response.send_message("Thread created")
+
+
+class DmCommunications(commands.Cog):
+    def __init__(self, bot: Rodhaj):
+        self._bot = bot
+        self._pool = bot.pool
+
+    @staticmethod
```

Comment:
There isn't a point in making it a staticmethod when it's only going to get internally used within the cog.